packages/pigeon/lib/dart_generator.dart

```diff
@@ -121,12 +121,12 @@ final BinaryMessenger$nullTag _binaryMessenger;
       }
       String argSignature = '';
       String sendArgument = 'null';
-      if (func.argType != 'void') {
-        argSignature = '${func.argType} arg';
+      if (func.argType.dataType != 'void') {
```

Comment:
I'm curious: why was avoiding LoD violations so important for `TypeDeclaration` that it needed a new interface wrapping its accessors, while this—which looks conceptually the same to me, and which you do throughout this PR—is not an issue for you?